src/backend_output_responder.cc

```diff
@@ -85,7 +88,24 @@ BackendOutputResponder::ProcessTensor(
       const int64_t* shape;
       TRITONBACKEND_InputProperties(
           input, nullptr, nullptr, &shape, nullptr, nullptr, nullptr);
+      if ((batchn_batch_size != -1) &&
+          ((batch_size_offset + shape[0]) > batchn_batch_size)) {
+        if (response != nullptr) {
+          RESPOND_AND_SET_NULL_IF_ERROR(
+              &response,
+              TRITONSERVER_ErrorNew(
+                  TRITONSERVER_ERROR_UNSUPPORTED,
+                  std::string(
+                      "failed to split the output tensor `" + output_name +
+                      "` in responses: expected batch size of atleast " +
+                      std::to_string(batch_size_offset + shape[0]) +
+                      " in model output, got " +
+                      std::to_string(batchn_batch_size))
+                      .c_str()));
+        }
+      }
       batchn_shape[0] = shape[0];
+      batch_size_offset += shape[0];
```

Comment:
Do we need this check specifically? Don't we already use `tensor_offset` for similar check?

Reply:
By overriding the batchn_shape[0] dimension, we are missing the total batch size of the input tensor. You are right that we can use tensor_offset to compute the batch_size_offset(tensor_offset divided by `GetByteSize(datatype, shape without batch size)`), but it will be a lot of computations and add overhead.

We don't have `batchn_byte_size`, computed before overrirding the batch dimension. `tensor_offset` is only used to get to a memory location from where the data for the response should be copied. Unfortunately, we don't have a check for that if it exceeds `batchn_byte_size`. The only case where it can exceed `batchn_byte_size` is when we override the batch dimension. To give out more specific/comprehensible errors, I stuck with `batch_sizes`

The current output looks like:

`failed to split the output tensor `probability` in responses: expected batch size of atleast 1 in model output, got 0`